rpc/src/v1/helpers/signature.rs

```diff
@@ -0,0 +1,27 @@
+use ethkey::{recover, public_to_address, Signature};
+use jsonrpc_core::Result;
+use v1::types::{Bytes, H520, RichBasicAccount, BasicAccount};
+use v1::helpers::errors;
+use v1::helpers::dispatch::eth_data_hash;
+use hash::keccak;
+
+pub fn verify_signature(is_prefixed: bool, message: Bytes, signature: H520, chain_id: Option<u64>) -> Result<RichBasicAccount> {
+	let mut hash = keccak(message.0.clone());
+	if is_prefixed {
+		hash = eth_data_hash(message.0);
+	}
```

Comment:
Since you are not using the message after you have computed the `hash` the ```clone``` seems needless!

Consider: 
```let hash = if is_prefixed { eth_data_hash(message.0) } else { keccak(message.0) }; ```

Also I think keccak supports slices so should be possible to pass-by-reference too!